lib/trip_dispatcher.rb

```diff
@@ -27,14 +41,58 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    #Wave 3: Find driver_by_status
+    def find_driver_status(status)
+      Driver.validate_id(status)
+      drivers.each do |driver|
+        if driver.status == :AVAILABLE
+          return driver
+        end
+      end
+      return nil
+    end
+
+    #Wave 3: request_trip method
+    #Wave 3: added new instance of trip
+    def request_trip(passenger_id)
+      available_driver = @drivers.select {|driver| driver.status == :AVAILABLE}.first
```

Comment:
Using `find` here is a little clearer (since it's easy to miss the `.first` at the end of the line):

```suggestion
      available_driver = @drivers.find {|driver| driver.status == :AVAILABLE}
```